### ISSUE TITLE
`Shift+Click` lost its functionality on the `Smart Reply` button

### ISSUE THREAD

**Issue by sphh** [caligraf/ChangeQuote#36]
In plain Thunderbird it is possible to switch the reply format by holding the `SHIFT` key, when clicking on the `Smart Reply` button: If the standard reply format is `Plain text`, holding the `SHIFT` key would switch to html format.

With `ChangeQuote` installed, this functionality is lost. Now I would have to click on the down-arrow to select the format. Would it be possible to retain the original behaviour?

I would also prefer a configuration option to remove the down-arrow on the `Smart Reply` button, but that is just a minor issue …

Thunderbird 91.11.0 (64-bit) on Linux Mint 20.3

**Comment by caligraf** [comment]
I have just released a new version compatible with Thunderbird 115.
https://github.com/caligraf/ChangeQuote/releases/download/1.7.6/ChangeQuote.xpi
The behavior of SHIFT key should be the same with and without ChangeQuote installed.
Could you retest with the version 1.7.6 ?

**Comment by sphh** [comment]
Many thanks for the update. I installed version 1.7.6 in Betterbird 115.3.2. In ChangeQuote's settings I ticked `Use in reply the same format (html or text) of the original mail` and `Default`. When I click on the `Smart Reply` button the reply is _always_ in plain text, regardless of the format of the message I reply to. If I Shift-click on the `Smart Reply` button I _always_ get the reply in HTML format.

What I meant in my original post is, that I would love that a simple click always opens the message in the format of the original message. A Shift-click would always open the reply in the other format. Is that possible? What settings would I need?

**Comment by sphh** [comment]
Same with Thunderbird 115.3.1 (64-bit) – both on Linux Mint 21.2 Cinnamon.

**Comment by caligraf** [comment]
if you tick `Use in reply the same format (html or text) of the original mail` and `Default`.  the email will be sent in the same format as the original email. But I confirm there is a bug when I reply in the compose window it is in plain text even if the original email is in HTML, I will fix it.
And if you ticked the setting `Compose messages in HTML format` in `Composition & Addressing` part of your Account settings the email will in HTML in the compose window even if you reply to an email in plain text. So you need to untick `Compose messages in HTML format`

**Comment by sphh** [comment]
I never had the `Compost messages in HTML format` ticked in account settings, but it might be worth to add a note to the ChangeQuote settings to avoid confusion.

Thanks for looking into this!

**Comment by caligraf** [comment]
Could you check the type of the format in source of the emails you try to reply with the same format ?

The source of the email can be displayed by clicking on `More` button and choose `View source`.
In the source the types of format are the lines Content-Type.
I would like to know if the problem happens for a particular type ( multipart/alternative, text/plain, text/html, multipart/mixed, ...  )

**Comment by sphh** [comment]
I now tried it on the following emails (all `Content-Type`s in the main header):
- Content-Type: multipart/related
- Content-Type: multipart/alternative
- Content-Type: text/plain
Both open the mail in plain text with a click and in HTML format with a Shift-Click.

**Comment by caligraf** [comment]
f you tick `Use in reply the same format (html or text) of the original mail` and `Default`you have below behavior:
- Content-Type: multipart/related : the reply email will be in plain text if the related content is in plain text , there should be another Content-type in the source of the email : it can be plain text, html or alternative
- Content-Type: multipart/alternative: the reply email will be in plain text if `Compose messages in HTML` is untick otherwise it will be in html
- Content-Type: text/plain: the reply email will be in plain text

To summarize  `Default` will use the value of `Compose messages in HTML`  in account settings: untick equals reply in plain text and tick equals reply in HTML

I will do a fix to handle the case of multipart/related, currently it is not handled correctly.

**Comment by caligraf** [comment]
It seems the account settings  `Compose messages in HTML` overrides the ChangeQuote settings. I need to check with Thunderbird team why.
